111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,229 @@
+    - `--tls-verify` - skip TLS verification for insecure registries.
+
+As for the Kaniko additional options, if the user-specified Buildah options will contain forbidden options (or not known), user will be notified by message inside the `.status` section of `KafkaConnect` resource, the `InvalidResourceException` will be thrown (logged inside the operator log) and the build will fail.
+
+If Buildah is enabled through the feature gate and the user provides Kaniko-specific options, those options will be ignored, and a warning will be reported in the `.status` section of the `KafkaConnect` CR.
+Once the feature gate will be moved to beta, the additional Kaniko options field will be deprecated.
+Additionally, we change the environment variable from `STRIMZI_DEFAULT_KANIKO_EXECUTOR_IMAGE` to `STRIMZI_DEFAULT_BUILDAH_IMAGE` in our deployment files.
+When the feature gate will be moved to GA and Kaniko implementation will be removed, the environment variable for Kaniko will be removed from the code as well (and ignored in case that user sets it).
+
+Finally, this feature will be available on Kubernetes only - it will not be available on OpenShift, which is mentioned as one of the rejected alternative.
+So the implementation and usage of OpenShift Build API will be kept.
+
+### YAML example of configuring Connect Build with Buildah options
+
+The following YAML example shows how the `KafkaConnect` CR with build configuration (and Buildah additional options) will look like:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+  annotations:
+    strimzi.io/use-connector-resources: "true"
+spec:
+  version: 4.1.0
+  replicas: 1
+  bootstrapServers: my-cluster-kafka-bootstrap:9093
+  tls:
+    trustedCertificates:
+      - secretName: my-cluster-cluster-ca-cert
+        pattern: "*.crt"
+  config:
+    group.id: connect-cluster
+    offset.storage.topic: connect-cluster-offsets
+    config.storage.topic: connect-cluster-configs
+    status.storage.topic: connect-cluster-status
+    config.storage.replication.factor: -1
+    offset.storage.replication.factor: -1
+    status.storage.replication.factor: -1
+  build:
+    output:
+      type: docker
+      image: my-internal-registry:5000/repository/image:latest
+      additionalBuildahBuildOptions:
+        - "--tls-verify=false"
+        - "--retry=4"
+      additionalBuildahPushOptions:
+        - "--tls-verify=false"
+        - "--quiet"
+    plugins:
+      - name: kafka-connect-file
+        artifacts:
+          - type: maven
+            group: org.apache.kafka
+            artifact: connect-file
+            version: 4.1.0
+```
+
+### Running Buildah in restricted environment
+
+The Buildah, similarly to Kaniko, has issues running in restricted environment.
+To run the build and push commands, it needs some capabilities like - `cap_kill`, `cap_setgid`, and `cap_setuid`.
```

Comment:
It is suprising push needs something special as that is just transfering some data to some remote registry.

Reply:
Well, i assumed you tried that it is the push which requires these and not the build?

Reply:
Well if it faield in the build then you should probably say that it is the build that needs it then.